### PR TITLE
complete permanent delete functionality

### LINKED ISSUE #94: Add favourites and trash functionality

### PR DESCRIPTION
**Pull request by 1298se**
resolve #94

### CODE REVIEW

File: contrail-api/functions/actions/resource.js [hunk 1]
```diff
@@ -271,6 +271,9 @@ checkUserPermissionsForResource = async (userId, resourceId) => {
     }
 }
 
+const getFileFromBucket =  (ownerId, resourceName) => {
+    return bucket.file(`${ownerId}/${resourceName}`);
```
Comment (virajg224) [inline contrail-api/functions/actions/resource.js]: this is not needed. its only one line

File: contrail-api/functions/actions/resource.js [hunk 2]
```diff
@@ -365,3 +358,45 @@ exports.downloadResourceZip = (req, res) => {
             }
         });
 }
+
+const deleteResource = async (userId, resource) => {
+    try {
+        const doc = await getResourceDocRef(resource.generation).get();
+        if (doc.exists) {
+            if (doc.data().permissions[userId] === "editor") {
+                console.log("iseditor");
```
Comment (virajg224) [inline contrail-api/functions/actions/resource.js]: console.logs

File: contrail-api/functions/actions/resource.js (hunk 1 above)
Comment (1298se) [reply contrail-api/functions/actions/resource.js]: What if we change the bucket path though!! :/

File: contrail-api/functions/actions/resource.js (hunk 2 above)
Comment (1298se) [reply contrail-api/functions/actions/resource.js]: resolved

**Pull request merged**